### PR TITLE
Makefile: switch to node_modules/ as submodule

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
Build and maintain the node_modules/ as a submodule, using the tools/node-modules script as it's done in the cockpit repo.

Updating the node_modules/ directory after changing package.json now involves running `tools/node-modules install` and then committing the result, and pushing the submodule commit with `tools/node-modules push`.

Copy the dependabot workflow to make sure things get properly rebuilt when our `node_modules` label is set (as dependabot does for us).

### CODE REVIEW

**Comment by jelly** [comment]
General recommendation, lets merge the icon PR first as it changes node_modules.

**Comment by allisonkarlitskaya** [comment]
> Lets merge the icon PR first as it changes node_modules.

Rebased.

File: Makefile [hunk 1]
```diff
@@ -7,7 +7,6 @@ TEST_OS = fedora-40
 endif
 export TEST_OS
 TARFILE=$(RPM_NAME)-$(VERSION).tar.xz
-NODE_CACHE=$(RPM_NAME)-node-$(VERSION).tar.xz
```
Comment (jelly) [inline Makefile:10]: Guess we can drop this in podman as well.

**Review by jelly — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (martinpitt) [inline Makefile:10]: What happened to this? We won't be able to get this into Fedora without rebuilding the bundle during rpm build.

File: packaging/cockpit-files.spec.in [hunk 2]
```diff
@@ -24,15 +23,10 @@ Requires: cockpit-bridge >= 316
 A filesystem browser for Cockpit
 
 %prep
-%autosetup -n %{name} -a 1
-# ignore pre-built bundle in release tarball and rebuild it
-# but keep it in RHEL/CentOS-8, as that has a too old nodejs
-%if ! 0%{?rhel} || 0%{?rhel} >= 9
-rm -rf dist
-%endif
```
Comment (martinpitt) [inline packaging/cockpit-files.spec.in:32]: Please put that back. This has nothing to do with node_modules/ as submodule.

**Pull request merged**